CTFd/utils/security/email.py

```diff
@@ -0,0 +1,40 @@
+from CTFd.cache import cache
+from CTFd.exceptions.email import (
+    UserConfirmTokenInvalidException,
+    UserResetPasswordTokenInvalidException,
+)
+from CTFd.utils.security.signing import hmac
+
+
+def generate_email_confirm_token(addr, timeout=1800):
+    nonce = hmac(addr)
```

Comment:
@ColdHeat this should not be hmac of address, I would suggest it to be a random number (more than 128bit secure entrophy). 

This is becuase otherwise a single password reset link can be reused infinitely by initiating reset function. 

Or to say it differently: the key is **the same** in every password reset for the same user. Stealing this token once allows to reset the password many times in the future.